### PR TITLE
Create restricted entry point for case management agents

### PR DESCRIPTION
**Pull request by peterdavidhamilton**
## Changes in this PR

- access to Supported first requires authenticating via Specify
- add a login page for caseworkers
- restrict access to case management via uid/email
- populate agents table using data from users table
- strip unnecessary code
- rephrase some specs
- add missing frozen string literal
- encapsulate caseworker authentication in an Rspec context
- make caseworker role filtering work using affiliation to an organisation 

## Screenshots of UI changes

`/support`
Screenshot 2021-09-22 at 11 46 42

## Next steps

- add SBP filtering at authentication using establishment type/category
- register uses with the proc ops team

### CODE REVIEW

File: app/controllers/support/agents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      #
+      if ::PagesController.dsi?
+        specify_user.update!(
+          email: session[:dfe_sign_in_uid],
+          first_name: session[:dfe_sign_in_uid],
+          last_name: session[:dfe_sign_in_uid],
+        )
+      end
+
+      if AGENT_UUIDS.include?(specify_user.dfe_sign_in_uid)
```
Comment (montychristo) [inline app/controllers/support/agents_controller.rb:20]: Do we also need a check to ensure we can login via DSI using the `AGENT_EMAILS` whitelist too?

File: config/initializers/case_workers.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# AGENT_EMAILS = %w[
+#   agent1@example.com
+#   agent2@example.com
+# ]
+
+# use in development when bypassing DSI
+#
+AGENT_UUIDS = %w[
```
Comment (montychristo) [inline config/initializers/case_workers.rb:8]: Should we make these environment variables?

File: config/routes.rb [hunk 3]
```diff
@@ -58,9 +58,10 @@
   #
   # Supported ------------------------------------------------------------------
   #
-  namespace :support do
-    get "admin", to: "admin#show"
+  get "support", to: "support/pages#start_page", as: :support_root
```
Comment (montychristo) [inline config/routes.rb:61]: We could use `root_path` inside the support namespace. Only a personal preference though

**Review by montychristo — DISMISSED** [review]
Couple of comments but mainly LGTM - nicely done! Also - a few code climate violations

File: app/controllers/support/agents_controller.rb (hunk 1 above)
Comment (peterdavidhamilton) [reply app/controllers/support/agents_controller.rb:20]: At the moment this code is bypassing DSI in development so we can change UIDs to filter on.
If we deployed that approach, you're right we'd want to instead be filtering emails, but it's not the only option and it wouldn't be sustainable.

The way we are going to do it now is to assign agents to a particular organisation; that way access control is in the hands of ProcOps and we don't need developer input going forward.

This PR needs the org filtering to be added before it's ready to merge.

File: config/initializers/case_workers.rb (hunk 2 above)
Comment (peterdavidhamilton) [reply config/initializers/case_workers.rb:8]: This file is going away once an organisation exists.

File: config/routes.rb (hunk 3 above)
Comment (peterdavidhamilton) [reply config/routes.rb:61]: I did do that originally and now I can't remember what prompted me to change it? Think there was a little gotcha and doing it this way made it more workable.

File: spec/features/support/case_agent_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  it "displays the submit button" do
+    expect(find_button("Assign")).to be_present
+  end
+
+  context "when assigning an agent to a case" do
+    before do
+      find("label.govuk-radios__label", text: "Procurement Specialist").click
+      find_button("Assign").click
```
Comment (peterdavidhamilton) [inline spec/features/support/case_agent_spec.rb:26]: @ClaudioCarmeli this is me renaming your `case_edit_spec` to focus on one specific editing feature, i.e. the assignment of a caseworker. I'm going to change this line to `click_button`, just drawing your attention to that method and this upcoming change.

**Review by EdwinKruglov — APPROVED** [review]

File: app/presenters/support/base_presenter.rb [hunk 5]
```diff
@@ -1,20 +1,22 @@
+# frozen_string_literal: true
+
 module Support
   class BasePresenter < SimpleDelegator
     # @return [String]
     def created_at
-      format_date(super)
+      super.strftime(date_format)
```
Comment (peterdavidhamilton) [inline app/presenters/support/base_presenter.rb:7]: @EdwinKruglov for your PR this is better

File: app/presenters/support/base_presenter.rb [hunk 6]
```diff
@@ -1,20 +1,22 @@
...
     end
 
   private
 
     # @return [String]
-    def format_date(date)
-      date.strftime("%e %B %Y")
+    def date_format
+      "%e %B %Y"
```
Comment (AndrewCowle) [inline app/presenters/support/base_presenter.rb:19]: date.strftime("%e %B %Y at %H:%M")

I changed this in my PR so we show times alongside (spoke to Adam and Robin, was some variance in the PT and we felt having time makes more sense) the change will go in with my PR, just wanted to raise it.

File: config/routes.rb (hunk 3 above)
Comment (AndrewCowle) [reply config/routes.rb:61]: happy with support_root - need to look at the namescape vs scope in route so it's better unique name

**Review by AndrewCowle — APPROVED** [review]

File: app/presenters/support/agent_presenter.rb [hunk 7]
```diff
@@ -1,8 +1,21 @@
...
     def full_name
       "#{first_name} #{last_name}"
     end
+
+    # @return [false]
+    def guest?
+      false
+    end
+
+    # TODO: use this in auth checks
```
Comment (Claudio-And) [inline app/presenters/support/agent_presenter.rb:15]: Question: why will `.agent` be needed in those check?

**Review by Claudio-And — APPROVED** [review]

**Review by EdwinKruglov — APPROVED** [review]

File: app/presenters/support/base_presenter.rb (hunk 6 above)
Comment (peterdavidhamilton) [reply app/presenters/support/base_presenter.rb:19]: good stuff if we have a few types we can put them in a hash and call them by name or something

File: spec/features/support/case_agent_spec.rb (hunk 4 above)
Comment (peterdavidhamilton) [reply spec/features/support/case_agent_spec.rb:27]: I did not make this change as promised @Claudio-And could you do that please?

**Pull request merged**